Cannot find module 'react-dom/test-utils' from 'act-compat.js'

<!--

* Please fill out this template with all the relevant information so we can
  understand what's going on and fix the issue. We appreciate bugs filed and PRs
  submitted!

* If your issue is regarding one of the query APIs (`getByText`,
  `getByLabelText`, etc), then please file it on the `dom-testing-library`
  repository instead. If you file it here it will be closed. Thanks :)

* Please make sure that you are familiar with and follow the Code of Conduct for
  this project (found in the CODE_OF_CONDUCT.md file).

We'll probably ask you to submit the fix (after giving some direction). If
you've never done that before, that's great! Check this free short video
tutorial to learn how: http://kcd.im/pull-request

-->

- `react-testing-library` version: 6.0.0
- `react` version: 15.4.1
- `node` version: 8.11.3
- `npm` (or `yarn`) version: 1.12.3

### Relevant code or config:

```js
import 'react-testing-library/cleanup-after-each';
```

### What you did:

Just imported the `react-testing-library/cleanup-after-each`

### What happened:

Got this error `Cannot find module 'react-dom/test-utils' from 'act-compat.js'`

### Reproduction:

<!--
If possible, please create a repository that reproduces the issue with the
minimal amount of code possible.

Template repo: https://github.com/alexkrolick/dom-testing-library-template

Or if you can, try to reproduce the issue in a Codesandbox. You can fork the one
here: https://codesandbox.io/s/5z6x4r7n0p
-->

### Problem description:

The `react-addons-test-utils` was moved to `react-dom/test-utils` in the version `15.5`, that path doesn't work in `react-dom <15.5`

### Suggested solution:

<!--
It's ok if you don't have a suggested solution, but it really helps if you could
do a little digging to come up with some suggestion of how to improve things.
-->

Import from `react-addons-test-utils` or `react-dom/lib/ReactTestUtils` for `react-dom 15.4.x` versions, see this line
https://github.com/kentcdodds/react-testing-library/blob/master/src/act-compat.js#L3

--- a/src/act-compat.js
+++ b/src/act-compat.js
@@ -1,6 +1,12 @@
 import React from 'react'
 import ReactDOM from 'react-dom'
-import {act as reactAct} from 'react-dom/test-utils'
+
+let reactAct
+try {
+  reactAct = require('react-dom/test-utils').act
+} catch (error) {
+  // ignore, this is to support old versions of react
+}
 
 // act is supported react-dom@16.8.0
 // so for versions that don't have act from test utils

